projects/netgrif-components-core/src/lib/navigation/navigation-double-drawer/abstract-navigation-double-drawer.ts

```diff
@@ -317,28 +326,31 @@ export abstract class AbstractNavigationDoubleDrawerComponent implements OnInit,
      * Current level is set to a lower number in order to set the left side menu.
      * */
     onBackClick(): void {
-        if (this._uriService.isRoot(this._currentNode)) return;
-        this.currentNode = this._currentNode.parent;
+        if (this.leftLoading$.isActive || this.rightLoading$.isActive || this._uriService.isRoot(this._currentNode)) {
+            return
+        }
+        this._uriService.activeNode = this._currentNode.parent;
     }
 
     onItemClick(item: NavigationItem): void {
-        this._uriService.activeNode = this._currentNode;
-        if (this.hasItemChildren(item)) {
+        if (this.hasItemChildren(item) && !this.leftLoading$.isActive && !this.rightLoading$.isActive) {
             const path = item.resource.immediateData.find(f => f.stringId === GroupNavigationConstants.ITEM_FIELD_ID_NODE_PATH)?.value
             this._uriService.getNodeByPath(path).subscribe(node => {
-                this.currentNode = node
+                this._uriService.activeNode = node
             }, error => {
                 this._log.error(error);
             });
+        } else {
+            this._uriService.activeNode = this._currentNode;
         }
     }
 
     hasItemChildren(item: NavigationItem): boolean {
-        return item.resource.immediateData.find(f => f.stringId === GroupNavigationConstants.ITEM_FIELD_ID_HAS_CHILDREN)?.value
+        return item.resource?.immediateData.find(f => f.stringId === GroupNavigationConstants.ITEM_FIELD_ID_HAS_CHILDREN)?.value
```

Comment:
it is enought to put question mark behind resource? or it should be also behind immediateData?

Reply:
It's enough in my opinion. In practice every resource contains some immediateData, because the Case instance can be only from one process, that has defined immediate fields. In theory if the resource somehow doesn't contain immediateData (it's undefined), it won't be displayed in double-drawer component (an error would be caught and item ignored while resolving such item). So such item won't be a subject for this hasItemChildren function